server/src/routes/v2/apps.js

```diff
@@ -64,4 +87,207 @@ module.exports = [
             })
         },
     },
+    {
+        method: 'POST',
+        path: '/v2/apps',
+        config: {
+            auth: 'token',
+            tags: ['api', 'v2'],
+            payload: {
+                maxBytes: 20 * 1024 * 1024, //20MB
+                allow: 'multipart/form-data',
+                parse: true,
+                output: 'stream',
+                multipart: true,
+            },
+            validate: {
+                payload: CreateAppModel.payloadSchema,
+            },
+            plugins: {
+                'hapi-swagger': {
+                    payloadType: 'form',
+                },
+            },
+        },
+        handler: async (request, h) => {
+            if (!canCreateApp(request, h)) {
+                throw Boom.unauthorized()
+            }
+
+            const { app, logo, file } = request.payload
+
+            if (!app || !logo || !file) {
+                throw Boom.badRequest('App, logo and zip file are all required')
+            }
+
+            const appJsonPayload = JSON.parse(app)
+            const appJsonValidationResult = CreateAppModel.def.validate(
+                appJsonPayload
+            )
+
+            if (appJsonValidationResult.error) {
+                throw Boom.badRequest(appJsonValidationResult.error)
+            }
+
+            const knex = h.context.db
+
+            let currentUser = null
+            let currentUserId = null
+            let isManager = false
+            try {
+                currentUser = await getCurrentUserFromRequest(request, knex)
+                currentUserId = currentUser.id
+
+                isManager = currentUserIsManager(request)
+            } catch (err) {
+                throw Boom.unauthorized('No user found for the request')
+            }
+
+            let appId = null
+            let versionId = null
+            let iconId = null
+
+            const trx = await knex.transaction()
```

Comment:
Likewise, I think we should move this logic to compose app-services? And use `knex.transaction(trx => )` API, rather than committing and aborting manually.